composer/helpers/v2/src/UpdateChecker.php

```diff
@@ -48,10 +50,22 @@ public static function getLatestResolvableVersion(array $args): ?string
             $io->loadConfiguration($config);
         }
 
+        $package = $composer->getPackage();
+
+        $versionParser = new VersionParser();
+
+        $constraint = $versionParser->parseConstraints($latestAllowableVersion); // your version constraint
+        $packageLink = new Link($package->getName(), $dependencyName, $constraint);
+
+        $requires = $package->getRequires();
+        $requires[$dependencyName] = $packageLink;
+
+        $package->setRequires($requires);
+
         $install = new Installer(
             $io,
             $config,
-            $composer->getPackage(),  // @phpstan-ignore-line
+            $package,  // @phpstan-ignore-line
```

Comment:
does this ignore still need to be there?

Reply:
I dint add that :) I will remove it.